pkg/conduit/runtime.go

```diff
@@ -227,6 +237,12 @@ func (r *Runtime) Run(ctx context.Context) (err error) {
 		multierror.ForEach(err, func(err error) {
 			r.logger.Err(ctx, err).Msg("provisioning failed")
 		})
+		if r.Config.Pipelines.ExitOnError {
+			r.logger.Warn(ctx).
+				Err(err).
+				Msg("Conduit will shut down due to a pipeline provisioning failure and 'exit on error' enabled")
+			t.Kill(cerrors.Errorf("shut down due to 'exit on error' enabled: %w", err))
```

Comment:
Is there a reason why we don't just return the error at this point? We know we want Conduit to stop at this point.

Reply:
I'm not sure I understand... Return the error from the handler function?

Reply:
This code is not in the handler.

Reply:
Ah, right. Sure, we can do a return here.